### PR TITLE
Pruthvi/bi rnn

### PR DESCRIPTION
**Pull request by pruthviIntel**
Added Support to fuse Bi directional RNN in single MKLDNN RNN kernel.

With no RNN optimization
[Function_0_000_N6ngraph4pass15LikeReplacementE.pdf](https://github.com/NervanaSystems/ngraph/files/2810192/Function_0_000_N6ngraph4pass15LikeReplacementE.pdf)

Optimization just with RNN cells Fusion.
[Function_0_004_N6ngraph7runtime3cpu4pass9RNNFusionE.pdf](https://github.com/NervanaSystems/ngraph/files/2810164/Function_0_004_N6ngraph7runtime3cpu4pass9RNNFusionE.pdf)

After Bi Directional RNN 
[Function_0_007_N6ngraph7runtime3cpu4pass16BiDirectionalRnnE.pdf](https://github.com/NervanaSystems/ngraph/files/2810165/Function_0_007_N6ngraph7runtime3cpu4pass16BiDirectionalRnnE.pdf)
Optimization.

```
[pthoreho@nervana-skx21 build_g++_debug]$ ./test/unit-test --gtest_filter="cpu_fusion.bi_rnn*"                                                                                                                                                                                             
Note: Google Test filter = cpu_fusion.bi_rnn*
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from cpu_fusion
[ RUN      ] cpu_fusion.bi_rnn_interpreter_vs_cpu
mkldnn_verbose,exec,concat,ref:any,undef,in:f32_blocked out:f32_nc,num:2,20x40,0.0100098
mkldnn_verbose,exec,concat,ref:any,undef,in:f32_blocked out:f32_nc,num:2,20x40,0.00488281
mkldnn_verbose,exec,concat,simple:any,undef,in:f32_nc out:f32_nc,num:2,2x10,54.416
mkldnn_verbose,exec,concat,simple:any,undef,in:f32_nc out:f32_nc,num:2,2x10,0.971191
mkldnn_verbose,exec,concat,simple:any,undef,in:f32_nc out:f32_nc,num:2,2x10,0.977051
mkldnn_verbose,exec,rnn,ref:any,forward_training,,alg:vanilla_lstm,l1d2mb1t2_ic10sc10oc10_wi10ws10,402.54
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_tnc out:f32_tnc,num:1,2x1x20,0.000976562
[       OK ] cpu_fusion.bi_rnn_interpreter_vs_cpu (2699 ms)
[----------] 1 test from cpu_fusion (2699 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (2699 ms total)
[  PASSED  ] 1 test.
```
alg:vanilla_lstm,l1**d2**mb1t2_ic10sc10oc10_wi10ws10,402.54, `d=2` specifies bi -directional

### CODE REVIEW

**Comment by pruthviIntel** [comment]
@jbobba can u review it?

**Comment by pruthviIntel** [comment]
@jbobba address review comments. please take a look. when u r free :)

File: src/ngraph/runtime/cpu/mkldnn_utils.hpp [hunk 1]
```diff
@@ -122,6 +122,14 @@ namespace ngraph
...
                     return true;
                 }
+
+                // TODO(pruthvi): Populate this enums based of addition of new MKLDNN RNN variants
+                enum rnntype
+                {
+                    vanilla_rnn,
+                    vanilla_gru,
+                    vanilla_lstm
+                };
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/mkldnn_utils.hpp]: Move enum to RNN op class instead of here.

**Comment by pruthviIntel** [comment]
@jbobba this time charm? :)

**Comment by pruthviIntel** [comment]
@jbobba can you approve, if it looks okay.?

**Pull request merged**